src/coreclr/tools/Common/TypeSystem/IL/Stubs/PInvokeILEmitter.cs

```diff
@@ -114,20 +118,36 @@ private static Marshaller[] InitializeMarshallers(MethodDesc targetMethod, Inter
                     parameterType = methodSig[i - 1];
                 }
 
-                marshallers[i] = Marshaller.CreateMarshaller(parameterType,
-                                                    parameterIndex,
-                                                    methodSig.GetEmbeddedSignatureData(),
-                                                    MarshallerType.Argument,
-                                                    parameterMetadata.MarshalAsDescriptor,
-                                                    direction,
-                                                    marshallers,
-                                                    interopStateManager,
-                                                    indexOffset + parameterMetadata.Index,
-                                                    flags,
-                                                    parameterMetadata.In,
-                                                    isHRSwappedRetVal ? true : parameterMetadata.Out,
-                                                    isHRSwappedRetVal ? false : parameterMetadata.Return
-                                                    );
+                if (runtimeMarshallingEnabled)
+                {
+                    marshallers[i] = Marshaller.CreateMarshaller(parameterType,
+                                                        parameterIndex,
+                                                        methodSig.GetEmbeddedSignatureData(),
+                                                        MarshallerType.Argument,
+                                                        parameterMetadata.MarshalAsDescriptor,
+                                                        direction,
+                                                        marshallers,
+                                                        interopStateManager,
+                                                        indexOffset + parameterMetadata.Index,
+                                                        flags,
+                                                        parameterMetadata.In,
+                                                        isHRSwappedRetVal ? true : parameterMetadata.Out,
+                                                        isHRSwappedRetVal ? false : parameterMetadata.Return
+                                                        );
+                }
+                else
+                {
+                    marshallers[i] = Marshaller.CreateDisabledRuntimeMarshallingMarshaller(
```

Comment:
Nit: the 3 "Marshal" subexpressions feel a bit too much:

```suggestion
                    marshallers[i] = Marshaller.CreateDisabledMarshaller(
```